webapp/views/App/views/Home/SurveyInfo/SamplingPolygonEditor/SamplingPolygonEditor.js

```diff
@@ -0,0 +1,262 @@
+          {i18n.t('samplingPolygonOptions.controlPointOffsetNorth')}
+        </label>
+        <Input
+          id="sampling-polygon-control-point-offset-north"
+          value={samplingPolygonObject.controlpoint_offset_north}
+          validation={getFieldValidation(samplingPolygonObject.controlpoint_offset_north)}
+          onChange={controlPointOffsetNorthOnChange}
+          readOnly={readOnly}
+        />
+      </div>
+    )
+
+    items.push(
+      <div className="form-item" key={'controlpointOffsetEastEditor'}>
+        <label className="form-label" htmlFor="survey-info-sampling-control-point-offset-east">
+          {i18n.t('samplingPolygonOptions.controlPointOffsetEast')}
+        </label>
+        <Input
+          id="sampling-polygon-control-point-offset-east"
+          value={samplingPolygonObject.controlpoint_offset_east}
+          validation={getFieldValidation(samplingPolygonObject.controlpoint_offset_east)}
+          onChange={constrolPointOffsetEastOnChange}
+          readOnly={readOnly}
+        />
+      </div>
+    )
+
+    return items
+  }
+
+  const shapeOnChange = (e) => {
+    samplingPolygonObject.isCircle = e.target.value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const radiusOnChange = (value) => {
+    samplingPolygonObject.radius = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const numberOfPointsCircleOnChange = (e) => {
+    samplingPolygonObject.number_of_points_circle = e.target.value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const lengthLatitudeOnChange = (value) => {
+    samplingPolygonObject.length_latitude = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const lengthLongitudeOnChange = (value) => {
+    samplingPolygonObject.length_longitude = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const numberOfPointsNorthOnChange = (value) => {
+    samplingPolygonObject.number_of_points_north = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const numberOfPointsEastOnChange = (value) => {
+    samplingPolygonObject.number_of_points_east = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const offsetNorthOnChange = (value) => {
+    samplingPolygonObject.offset_north = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const offsetEastOnChange = (value) => {
+    samplingPolygonObject.offset_east = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const controlPointOffsetNorthOnChange = (value) => {
+    samplingPolygonObject.controlpoint_offset_north = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
+  const constrolPointOffsetEastOnChange = (value) => {
+    samplingPolygonObject.controlpoint_offset_east = value
+    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
+  }
```

Comment:
since the only part that changes is the "key" of the property in the samplingPolygonObject, I would do it in a different way.
I would create an array like this:
const samplingPolygonProperties = [{ key: 'length_latitude´, labelKey: 'lengthLatitude' }, ...]

then to render the items: 
samplingPolygonProperties.map(({key, labelKey}) => {
const inputId = `survey-info-sampling-control-point-${key}`
return <div className="form-item" key={key}>
        <label className="form-label" htmlFor={inputId}>
          {i18n.t(`samplingPolygonOptions.${labelKey}`)}
        </label>
        <Input
          id={inputId}
          value={samplingPolygonObject[key]}
          validation={getFieldValidation(samplingPolygonObject[key]}
          onChange={value => onPropertyChange({key, value})}
          readOnly={readOnly}
        />
      </div>
)

and the onPropertyChangeFunction will be generic, like: 
 const onPropertyChange = ({key, value}) => {
    samplingPolygonObject[key] = value
    setSamplingPolygon(JSON.stringify(samplingPolygonObject))
  }

another question: could the sampliing polygon object be put into the survey properties as an object instead? why is it "stringified"?

Reply:
Yeah correct. Made them own functions because I thought they would need to be different and forgot to change it in the end when it was apparent that they are the same. It was stringified since when I put it to the db the methods stringified it and I thought that your bindings always stringify them but found that it is not true and have changed it now.